frontend/src/components/header.js

```diff
@@ -42,65 +44,77 @@ BootstrapNavLink.propTypes = {
  * @param {object[]} props.routes
  * @returns
  */
+
 export function Header(props) {
     const { routes = [], infoComponents = null } = props;
+    let match = useRouteMatch("/:mainRoute/:subRoute?");
+    const { mainRoute } = match.params;
 
     if (routes.length === 0) {
         return <div>No Routes in Header</div>;
     }
 
+    const activeMainRoutes = routes.map((route) => (
+        <BootstrapNavItem
+            eventKey={route.route}
+            to={route.route}
+            key={route.route}
+            className="primary"
+        >
+            <h5>{route.name}</h5>
+        </BootstrapNavItem>
+    ));
+
+    const activeSubRoutes = routes
```

Comment:
This is really confusing...Maybe `availableSubRoutes` is a better name. Also, add a comment explaining what/why is going on.